src/js/components/ChannelNav/ChannelNav.js

```diff
@@ -0,0 +1,136 @@
+import PropTypes from 'prop-types';
+import React, { useContext, useState } from 'react';
+import { useHistory, useParams } from 'react-router';
+import '../../../scss/components/ChannelNav/ChannelNav.scss';
+import { GlobalContext } from '../../contexts/GlobalContextWrapper';
+import useDiscordData from '../../hooks/useDiscordData';
+import ChannelNavButton, { ChannelNavParent } from '../ChannelNavButton/ChannelNavButton';
+import UserControlPanel from '../UserControlPanel/UserControlPanel';
+
+const ChannelNav = () => {
+  const history = useHistory();
+  const { guild } = useParams();
+
+  const { Config } = useContext(GlobalContext);
+  const { discordAPIResources: { userGuilds, pending } } = Config.get(['discordAPIResources']);
+
+  let guildName = '';
+  let guildId = pending;
+
+  const guildData = useDiscordData(userGuilds);
+
+  if (guildData) {
+    // Checks that the URL param guild id matches the discordAPI user guild id
+    const findGuild = guildData.find((e) => e.id === guild);
+
+    if (findGuild) {
+      if (guild !== findGuild.id) {
```

Comment:
```suggestion
    if (findGuild && guild !== findGuild.id) {
```

Reply:
I will get an error when findGuild is not an object, I will use typeof Instead to make it more declarative